Sources/AppcuesKit/Presentation/Actions/Appcues/AppcuesTrackAction.swift

```diff
@@ -10,27 +10,67 @@ import Foundation
 
 @available(iOS 13.0, *)
 internal class AppcuesTrackAction: AppcuesExperienceAction {
-    struct Config: Decodable {
+    struct Config {
         let eventName: String
+        let attributes: [String: Any]?
     }
 
     static let type = "@appcues/track"
 
     private weak var appcues: Appcues?
 
     let eventName: String
-
+    let attributes: [String: Any]?
+    
     required init?(configuration: AppcuesExperiencePluginConfiguration) {
         self.appcues = configuration.appcues
 
         guard let config = configuration.decode(Config.self) else { return nil }
         self.eventName = config.eventName
+        self.attributes = config.attributes
     }
 
     func execute(completion: ActionRegistry.Completion) {
         guard let appcues = appcues else { return completion() }
 
-        appcues.track(name: eventName)
+        appcues.track(name: eventName, properties: attributes)
         completion()
     }
 }
+
+@available(iOS 13.0, *)
+extension AppcuesTrackAction.Config: Decodable {
+    // Custom decoding for this one - we want to just gather up all the key/value
+    // pairs from the trait config as raw user property values. We trim the set to
+    // only those of supported data types and then store in the resulting properties
+    // dictionary
+    init(from decoder: Decoder) throws {
+               
+        let container = try decoder.container(keyedBy: DynamicCodingKeys.self)
+
+        self.eventName = try container.decode(String.self, forKey: DynamicCodingKeys(stringValue: "eventName")!)
```

Comment:
FYI instead of `DynamicCodingKeys(stringValue: "eventName")!`, you can use `DynamicCodingKeys(key: String)` to avoid the force unwrap. If you look at the implementation, it does the same thing, but just isn't an optional init since the optional init is only required by the `CodingKey` contract.

And technically since `eventName` and `attributes` are known keys, you should probably do this instead:
```swift
enum CodingKeys: CodingKey {
    case eventName
    case attributes
}
...
let container = try decoder.container(keyedBy: CodingKeys.self)
self.eventName = try container.decode(String.self, forKey: .eventName)
```